src/form/acp.jl

```diff
@@ -291,3 +328,197 @@ function constraint_tp_oltc_flow(pm::PMs.GenericPowerModel{T}, nw::Int, i::Int,
         )
     end
 end
+
+
+"Links the power flowing into both windings of a variable tap transformer."
+function constraint_tp_trans_flow_var(pm::PMs.GenericPowerModel, i::Int, f_bus::Int, t_bus::Int, f_idx, t_idx, Ti_fr, Ti_im; nw::Int=pm.cnw)
+    # for ac formulation, indentical to fixed tap
+    constraint_tp_trans_flow(pm, i, f_bus, t_bus, f_idx, t_idx, Ti_fr, Ti_im)
+end
+
+
+function constraint_load_power_setpoint_wye(pm::PMs.GenericPowerModel{T}, nw::Int, cnd::Int, load_id::Int, pd::Real, qd::Real) where T <: PMs.AbstractACPForm
+    JuMP.@constraint(pm.model, PMs.var(pm, nw, cnd, :pd, load_id)==pd)
+    JuMP.@constraint(pm.model, PMs.var(pm, nw, cnd, :qd, load_id)==qd)
+end
+
+
+function constraint_load_power_prop_vm_wye(pm::PMs.GenericPowerModel{T}, nw::Int, cnd::Int, load_id::Int, load_bus_id::Int, scale_p::Real, scale_q::Real) where T <: PMs.AbstractACPForm
+    pd = PMs.var(pm, nw, cnd, :pd, load_id)
+    qd = PMs.var(pm, nw, cnd, :qd, load_id)
+    vm = PMs.var(pm, nw, cnd, :vm, load_bus_id)
+    JuMP.@NLconstraint(pm.model, pd==scale_p*vm)
+    JuMP.@NLconstraint(pm.model, qd==scale_q*vm)
+end
+
+
+function constraint_load_power_prop_vmsqr_wye(pm::PMs.GenericPowerModel{T}, nw::Int, cnd::Int, load_id::Int, load_bus_id::Int, cp::Real, cq::Real) where T <: PMs.AbstractACPForm
+    pd = PMs.var(pm, nw, cnd, :pd, load_id)
+    qd = PMs.var(pm, nw, cnd, :qd, load_id)
+    vm = PMs.var(pm, nw, cnd, :vm, load_bus_id)
+    JuMP.@NLconstraint(pm.model, pd==cp*vm^2)
+    JuMP.@NLconstraint(pm.model, qd==cq*vm^2)
+end
+
+
+"""
+For a delta load, sd = (s_ab, s_bc, s_ca), but we want to fix s = (s_a, s_b, s_c)
+s is a non-linear transform of v and sd, s=f(v,sd)
+s_a = v_a*conj(s_ab/(v_a-v_b) - s_ca/(v_c-v_a))
+s_b = v_b*conj(s_ab/(v_a-v_b) - s_ca/(v_c-v_a))
+s_c = v_c*conj(s_ab/(v_a-v_b) - s_ca/(v_c-v_a))
+"""
+function constraint_tp_load_power_setpoint_delta(pm::PMs.GenericPowerModel{T}, nw::Int, load_id::Int, load_bus_id::Int, pd::PMs.MultiConductorVector, qd::PMs.MultiConductorVector) where T <: PMs.AbstractACPForm
```

Comment:
This function should have an assertion checking the number of conductors/phases in the network model.

Reply:
I added this at the template level in constraint.jl